cmd/podman/compose.go

```diff
@@ -85,20 +86,19 @@ func composeProvider() (string, error) {
 		return "", errors.New("no compose provider specified, please refer to `man podman-compose` for details")
 	}
 
+	var lookupErrors []error
 	for _, candidate := range candidates {
 		path, err := exec.LookPath(os.ExpandEnv(candidate))
-		if err != nil {
-			if errors.Is(err, os.ErrNotExist) {
-				continue
-			}
-			return "", err
+		if err == nil {
+			// First specified provider "candidate" wins.
+			logrus.Debugf("Found compose provider %q", path)
+			return path, nil
 		}
-		// First specified provider "candidate" wins.
-		logrus.Debugf("Found compose provider %q", path)
-		return path, nil
+		logrus.Debugf("Error looking up compose provider %q: %v", path, err)
+		lookupErrors = append(lookupErrors, err)
```

Comment:
Do we care about ErrNotExists?

Reply:
Yes.  The idea was to also return those errors at the end, such that the users can see all configured providers with the associated errors, see https://github.com/containers/podman/pull/20508#discussion_r1374405011

Reply:
I have a question it assumes that the path of the candidate is in my $PATH ?

because for example if I install docker, I do have `$HOME/.docker/cli-plugins/docker-compose` but it's never found by `podman compose`

https://github.com/containers/common/blob/7c7a603ba11f590056fa0fbb4a8ec8ba9e9fd97d/pkg/config/default_darwin.go#L27

or it's because $HOME is not resolved ?

Reply:
With this change it should work.

Each item gets expanded so $HOME will resolve. `LookPath` will not consult `$PATH` if the passed string contains a slash.

Reply:
ok, tried without the fix = error

and with the fix, it's finding the binary 👍

Reply:
Thanks for testing :pray:

Reply:
with the patch:

```
 ./bin/darwin/podman compose --version
 >>>> Executing external compose provider "/Users/benoitf/.docker/cli-plugins/docker-compose". Please refer to the documentation for details. <<<<

Docker Compose version v2.20.2-desktop.1
```

without:

```
$ podman compose --version
Error: exec: "docker-compose": executable file not found in $PATH
```